packages/edit-site/src/components/block-editor/index.js

```diff
@@ -226,3 +265,26 @@ export default function BlockEditor() {
 		</ExperimentalBlockEditorProvider>
 	);
 }
+
+/**
+ * Returns the appropriate block editor state for a given entity type.
+ *
+ * @param {string} kind the entity kind
+ * @param {string} type the entity type
+ * @return {[WPBlock[], Function, Function]} The block array and setters.
+ */
+function useEntitiyTypeBlockEditor( kind, type ) {
+	const entityId = useEntityId( kind, type );
+
+	const entityBlockEditor = useEntityBlockEditor( kind, type, {
+		id: entityId,
+	} );
+
+	const navigationEditorBlockEditor = useNavigationBlockEditor( entityId );
+
+	if ( type === `wp_navigation` ) {
+		return navigationEditorBlockEditor;
+	}
+
+	return entityBlockEditor;
+}
```

Comment:
The logic in this component is becoming more complex. It seems basically this component is a way to instantiate a "block editor" in the "site editor" and that the config (provider props) depends on the edited entity. So I wonder if there's a way to reorganize this component a bit by moving the entity specific logic to dedicated component. One option could be having three components: TemplateBlockEditorProvider, TemplatePartBlockEditorProvider and NavigationBlockEditorProvider instead of `BlockEditorProvider` and switch the one used depending on the active entity.

I'm not sure it's the best approach but I'm thinking we should try to make this component easier to follow and allow all these "if"s everywhere and weird hooks.